### ISSUE TITLE
Update: allowed roles for img elements

### ISSUE THREAD

**Issue by scottaohara** [validator/validator#1599]
[ARIA in HTML](https://github.com/w3c/html-aria/pull/453) is being updated to clarify allowed use of roles on img elements.  This was changed a couple years back, to purposefully limit the allowed roles on the element - but in doing so the rules were too closely tied to the presence/absence of alternative text. The rules were written to align with that requirement, but in doing so, we failed to account for the fact that attributes such as `title`, `aria-label` and `aria-labelledby` can provide images accessible names, and regardless of the HTML validation error in not providing an alt, if one of those attributes is specified then the image will have an accessible name, and thus the roles that are valid for when the img has an alt are also properly exposed.

When an image has an accessible name (provided by either alt, aria-label, aria-labelledby or title attributes) the following roles are allowed:
button, checkbox, link, menuitem, menuitemcheckbox, menuitemradio, meter, option, progressbar, radio, scrollbar, separator, slider, switch, tab or treeitem. (img is also allowed, but NOT RECOMMENDED.)
And the dpub role `doc-cover`

Note: `meter` is a new allowed role on the element - as it is similar to `proressbar` which is already allowed.

For images lacking an alt attribute and have no `title`, `aria-label` or `aria-labelledby` attributes:
role=none or presentation are allowed
role=img is also allowed, but NOT RECOMMENDED.

For images with an alt="" and no `aria-label` or `aria-labelledby` attributes:
role=none or presentation are allowed, but NOT RECOMMENDED.

[Test cases for the passing examples](https://w3c.github.io/html-aria/tests/img-allowed-roles.html)

**Comment by eshellman** [comment]
More than a year later, the validator does not accept the markup with role="presentation" recommended here: https://www.w3.org/WAI/tutorials/images/decorative/

Instead it complains: `An “img” element must have an “alt” attribute, except under certain conditions. For details, consult guidance on providing text alternatives for images.`

Or, with the alt attribute, it complains `An “img” element which has an “alt” attribute whose value is the empty string must not have a “role” attribute.`

Tools being developed to mitigate html inaccessible because of empty alt text need to be able to distinguish when the the alt attribute in intentionally empty, or whether, as in 90% of our 70,000 files, are only empty to satisfy the validator.

Please prioritize fixing this bug.

**Comment by pkra** [comment]
I was running into this today for a contrived example:

```html
<img src="#" aria-labelledby="id">
<span hidden id="id">Alttext</span>
```

From a quick check, UAs seem to consistently expose this as per spec (role=image, name "Alttext").

Validation gave me two errors.

> Error: An img element with no alt attribute must not have any aria-* attributes other than aria-hidden.
> Error: An img element must have an alt attribute, except under certain conditions. For details, consult [guidance on providing text alternatives for images](https://www.w3.org/WAI/tutorials/images/).

**Comment by pkra** [comment]
I think this is the same as #1198.
